### DISCUSSION TITLE
Is there a way to use pydantic manually (without BaseModel)

### DISCUSSION THREAD

**Discussion by erezinman** [pydantic/pydantic#2556] [Question] [answered]
Hi, 
in order to check dynamic fields outside a pydantic `BaseModel`, I need a function that simply checks a value against an annotation at runtime. Something along the lines of `pydantidc.is_valid(value, typing_annotation_or_type)`. Is there such a thing?

Thanks.

**Comment by PrettyWood** [comment, marked as answer]
Hi @erezinman
If you want to keep the default behaviour of pydantic with coercion, there you can use the `parse_obj_as` util
```py
from pydantic import ValidationError, parse_obj_as

def is_valid(value, type_):
    try:
        parse_obj_as(type_, value)
        return True
    except ValidationError:
        return False

assert is_valid((1, 'q'), list[str])
```

If you want strictness, for now it's probably easier to use another lib like https://github.com/PrettyWood/typingx. Note there are probably better libs out there than mine.

Hope it helps

**Reply by erezinman** [reply]
Thank you - I'll try it. I already found a "hack" by defining a `ModelField` and validating it. The code in this repository is really pretty.
Regardless, I suggest that you add an option to the `BaseConfig` of `strict_parsing` to make the underlying mechanics map all the `int`s to `StrictInt`s, `str`s to `StrictStr`s, etc.

The way I handled it is by doing something like the following (integrating it to your code):
```python
def is_valid(value, type_):
    try:
        converted_value = parse_obj_as(type_, value)
        return converted_value == value
    except ValidationError:
        return False

```

I'll also have a look at your repository. Thanks!
